source/Board.cpp

```diff
@@ -554,5 +535,86 @@ bool Board::isKnightMove(const std::pair<int, int>& fromCoords, const std::pair<
 
 }
 
+bool Board::isValidMove(const std::pair<int, int> &fromCoords, const std::pair<int, int> &toCoords) const
+{
+	std::cout << "Attempting to move from " << intToAlgebraic(fromCoords) << " to " << intToAlgebraic(toCoords) << '\n';
+
+	// Check if from and to locations are on the board
+	if (!isOnBoard(fromCoords) || !isOnBoard(toCoords))
+	{
+		std::cout << "Error: Coords are out of bounds" << '\n';
+		return false;
+	}
+
+	// Check if from location is occupied by piece of same color
+	if (isOccupiedSameColor(fromCoords, toCoords))
+	{
+		std::cout << "Error: Cannot move from " << intToAlgebraic(fromCoords) << " to " << intToAlgebraic(toCoords)
+				  << " because there's a friendly piece at " << intToAlgebraic(toCoords) << '\n';
+		return false;
+	}
+
+	// Check if this is a valid move for this piece
+	const Piece *piece = getPiece(fromCoords);
```

Comment:
```suggestion
	const Piece *piece = getPiece(fromCoords);
	if (piece == nullptr)
	{
		std::cout << "Error: No piece at this starting square" << '\n';
		return false;
	}
```

Reply:
I noticed that I can crash the game by inputting something like "e6 e8" when there's no piece at e6. Easy to fix with a small change like this.